docs/examples/intro_to_kernels.py

```diff
@@ -0,0 +1,328 @@
+
+x = jnp.linspace(-3.0, 3.0, num=200).reshape(-1, 1)
+
+meanf = gpx.mean_functions.Zero()
+
+for k, ax in zip(kernels, axes.ravel()):
+    prior = gpx.Prior(mean_function=meanf, kernel=k)
+    rv = prior(x)
+    y = rv.sample(seed=key, sample_shape=(10,))
+    ax.plot(x, y.T, alpha=0.7)
+    ax.set_title(k.name)
+
+
+# %% [markdown]
+# It should be noted that commonly used Matérn kernels use half-integer values of $\nu$, such as $\nu = 1/2$ or $\nu = 5/2$. The fraction is sometimes omitted when naming the kernel, so that $\nu = 1/2$ is referred to as the Matérn12 kernel, and $\nu = 5/2$ is referred to as the Matérn52 kernel.
+#
+# The plots above clearly show that the choice of $\nu$ has a large impact on the *smoothness* of the functions being modelled by the GP, with functions drawn from GPs defined with the Matérn kernel becoming increasingly smooth as $\nu \to \infty$. More formally, this notion of smoothness is captured through the mean-square differentiability of the function being modelled. Functions sampled from GPs using a Matérn kernel are $k$-times mean-square differentiable, if and only if $\nu > k$. For instance, functions sampled from a GP using a Matérn12 kernel are zero times mean-square differentiable, and functions sampled from a GP using the RBF kernel are infinitely mean-square differentiable.
+#
+# As an important aside, a general property of the Matérn family of kernels is that they are examples of *stationary* kernels. This means that they only depend on the *displacement* of the two points being compared, $\mathbf{x} - \mathbf{x}'$, and not on their absolute values. This is a useful property to have, as it means that the kernel is invariant to translations in the input space. They also go beyond this, as they only depend on the Euclidean *distance* between the two points being compared, $|\mathbf{x} - \mathbf{x}'|$. Kernels which satisfy this property are known as *isotropic* kernels. This makes the function invariant to all rigid motions in the input space, such as rotations.
+
+# %% [markdown]
+# ## Inferring Kernel Hyperparameters
+
+# %% [markdown]
+# Most kernels have several *hyperparameters*, which we denote $\mathbf{\theta}$, which encode different assumptions about the underlying function being modelled. For the Matérn family descibred above, $\mathbf{\theta} = \{\nu, \kappa, \sigma\}$. A fully Bayesian approach to dealing with hyperparameters would be to place a prior over them, and marginalise over the posterior derived from the data in order to perform predictions. However, this is often computationally very expensive, and so a common approach is to instead *optimise* the hyperparameters by maximising the log marginal likelihood of the data. Given training data $\mathbf{D} = (\mathbf{X}, \mathbf{y})$, assumed to contain some additive Gaussian noise $\epsilon \sim \mathcal{N}(0, \sigma^2)$, the log marginal likelihood of the dataset is defined as:
```

Comment:
Maybe note in gpjax that `\nu` is fixed and not something that can be inferred through optimisation.